cli/cmd/install.go

```diff
@@ -834,61 +838,32 @@ func (options *installOptions) proxyConfig() *pb.Proxy {
 	}
 }
 
-// exitIfClusterExists checks the kubernetes API to determine
-// whether a config exists and exits if it does exist or if an error is
-// encountered.
-//
-// This bypasses the public API so that public API errors cannot cause us to
-// misdiagnose a controller error to indicate that no control plane exists.
-func exitIfClusterExists() {
-	k, err := k8s.NewAPI(kubeconfigPath, kubeContext, 0)
-	if err != nil {
-		fmt.Fprintln(os.Stderr, "Unable to build a Kubernetes client to check for configuration. If this expected, use the --ignore-cluster flag.")
-		fmt.Fprintf(os.Stderr, "Error: %s\n", err)
-		os.Exit(1)
+func globalResourcesExist() bool {
+	checks := []healthcheck.CategoryID{
+		healthcheck.LinkerdPreInstallGlobalResourcesChecks,
 	}
 
-	c := k.CoreV1().ConfigMaps(controlPlaneNamespace)
-	if _, err = c.Get(k8s.ConfigConfigMapName, metav1.GetOptions{}); err != nil {
-		if kerrors.IsNotFound(err) {
-			return
-		}
-
-		fmt.Fprintln(os.Stderr, "Unable to build a Kubernetes client to check for configuration. If this expected, use the --ignore-cluster flag.")
-		fmt.Fprintf(os.Stderr, "Error: %s\n", err)
-		os.Exit(1)
-	}
+	hc := healthcheck.NewHealthChecker(checks, &healthcheck.Options{
+		ControlPlaneNamespace: controlPlaneNamespace,
+		KubeConfig:            kubeconfigPath,
+	})
 
-	fmt.Fprintln(os.Stderr, "Linkerd has already been installed on your cluster in the linkerd namespace. Please run upgrade if you'd like to update this installation. Otherwise, use the --ignore-cluster flag.")
-	os.Exit(1)
+	notExist := hc.RunChecks(func(result *healthcheck.CheckResult) {})
+	return !notExist
```

Comment:
I got a bit confused here with the `notExist` variable. 

When we run `hc.RunChecks()` here does a `false` return mean that global resources already exist? And that is why you are returning `!notExist`? I have the same question about `linkerdConfigConfigMapExists()`.

A longer but more explicit way could be:

```diff
diff --git a/cli/cmd/install.go b/cli/cmd/install.go
index 701d63ee..c078f2a7 100644 func globalResourcesExist() bool {
+       globalResourcesAlreadyExist := false // default value
+
        checks := []healthcheck.CategoryID{
                healthcheck.LinkerdPreInstallGlobalResourcesChecks,
        }
@@ -848,8 +850,13 @@ func globalResourcesExist() bool {
                KubeConfig:            kubeconfigPath,
        })

-       notExist := hc.RunChecks(func(result *healthcheck.CheckResult) {})
-       return !notExist
+       checksPass := hc.RunChecks(func(result *healthcheck.CheckResult) {})
+
+       if !checksPass {
+               globalResourcesAlreadyExist = true
+       }
+
+       return globalResourcesAlreadyExist
 } 
```

Or could you simplify by returning the result of `hc.RunChecks()` directly and changing the boolean wherever you are expecting that result?